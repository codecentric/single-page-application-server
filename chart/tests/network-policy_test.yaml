suite: test network-policy
templates:
  - network-policy.yaml
tests:
  - it: should not render network-policy if disabled
    values:
      - ./values/minimal_values.yaml
    set:
      networkPolicy.enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should render network-policy with egress isolation by default
    values:
      - ./values/minimal_values.yaml
    asserts:
      - matchSnapshot: { }
  - it: should render network-policy with custom egress policy
    values:
      - ./values/minimal_values.yaml
    set:
      networkPolicy:
        egress:
          - to:
              - ipBlock:
                  cidr: 10.0.0.0/24
            ports:
              - protocol: TCP
                port: 5978
    asserts:
      - matchSnapshot: { }
  - it: should render network-policy with custom ingress policy
    values:
      - ./values/minimal_values.yaml
    set:
      networkPolicy:
        policyTypes:
          - Ingress
        ingress:
          - from:
              - namespaceSelector:
                  matchLabels:
                    "kubernetes.io/metadata.name": myingresscontroller
            ports:
              - protocol: TCP
                port: http
              - protocol: TCP
                port: https
    asserts:
      - matchSnapshot: { }
