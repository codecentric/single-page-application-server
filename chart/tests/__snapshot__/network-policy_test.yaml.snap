should render network-policy with custom egress policy:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      egress:
        - ports:
            - port: 5978
              protocol: TCP
          to:
            - ipBlock:
                cidr: 10.0.0.0/24
      ingress: []
      podSelector:
        matchLabels:
          app.kubernetes.io/instance: default
          app.kubernetes.io/name: RELEASE-NAME
      policyTypes:
        - Egress
should render network-policy with custom ingress policy:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      egress: []
      ingress:
        - from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: myingresscontroller
          ports:
            - port: http
              protocol: TCP
            - port: https
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/instance: default
          app.kubernetes.io/name: RELEASE-NAME
      policyTypes:
        - Ingress
should render network-policy with egress isolation by default:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      egress: []
      ingress: []
      podSelector:
        matchLabels:
          app.kubernetes.io/instance: default
          app.kubernetes.io/name: RELEASE-NAME
      policyTypes:
        - Egress
